ui/react-components/components/GridSummary/GridSummary.jsx

```diff
@@ -23,14 +25,83 @@ const transformGridData=(gridData)=>{
     };
   });
 }
+export const transformData = (gridData)=>{
+  let rowMap = []
+  for(let data in gridData){
+    const rowList = []
+    for(let rowData in gridData[data]){
+      rowList.push({
+        date: rowData,
+        uuids: gridData[data][rowData].uuids,
+        count: gridData[data][rowData].count
+      })
+    }
+    rowMap.push({rowLabel:data, rowDataList: rowList})
+  }
+  return sortBy(rowMap, row => row.rowLabel)
+}
+export const setMap = ( map, date, name, uuid) => {
+  if(!map[name]){
+    map[name]={}
+  }
+  if(!map[name][date]){
+    map[name][date]= {
+      uuids: [],
+      count: 0
+    }
+  }
+  map[name][date].uuids.push(uuid);
+  map[name][date].count += 1
+}
+
+export const transformAppointmentsData = (data) => {
+  const specialityMap = {}
+  const providerMap = {}
+  const locationMap = {}
+  for(let element in data) {
+    const {service, startDateTime, provider, providers, location} = data[element]
+    const { uuid, speciality } = service
+    const date = moment(startDateTime).format("YYYY-MM-DD")
+    setMap(specialityMap, date, speciality.name, uuid)
+    if(provider){
+      setMap(providerMap, date, provider.name, uuid)
+    }
+    else{
+      for(let i in providers){
+        setMap(providerMap, date, providers[i].name, uuid)
+      }
+    }
+    if(location) {
+      setMap(locationMap, date, location.name, uuid)
+    }
+  }
+  return [transformData(specialityMap),
+  transformData(providerMap),
+  transformData(locationMap)]
+}
 
 const GridSummary = props => {
-  const { gridData=[], weekStartDate = moment().startOf("isoweek"), onClick } = props;
+  const { gridData=[], weekStartDate = moment().startOf("isoweek"), onClick, gridName } = props;
   let week = []
-  const gridSummaryData=transformGridData(gridData)
+  const { fullSummary } = React.useContext(AppContext)
+  if(gridData.length === 0 && fullSummary){
+    return (
+        <div className={classNames(tableGridWrapper)}>
+          <h3 className={gridHeading}>{gridName}</h3>
+          <table className={tableGridSummary}>
+            <tbody>
+              <tr>
+                <td></td>
+                <td className={noAppointments}>This week, no appointment is available for any of the {gridName}</td>
```

Comment:
@Phanindra-tw this should be a label so that it can be configured based on the locale.